### ISSUE TITLE
error running cat_per_objec

### ISSUE THREAD

**Issue by jamg-upv** [jmgirard/agreement#2]
I get this error running this code 
```r
 library(agreement)
 data(ordered)
 results1 <- cat_per_object(ordered, categories = c(0, 1, 2, 3), weighting = "linear")
```
The log
<error/vctrs_error_subscript_oob>
Can't subset columns that don't exist.
x Column `linear` doesn't exist.
Backtrace:
  1. agreement::cat_per_object(...)
  2. agreement:::prep_data_cat(.data, categories, weighting, warnings)
  4. dplyr:::[http://select.data](https://t.co/YsoAdh97qY).frame(...)
  5. tidyselect::eval_select(expr(c(...)), .data)
  6. tidyselect:::eval_select_impl(...)
 14. tidyselect:::vars_select_eval(...)
 15. tidyselect:::walk_data_tree(expr, data_mask, context_mask)
 16. tidyselect:::eval_c(expr, data_mask, context_mask)
 17. tidyselect:::reduce_sels(node, data_mask, context_mask, init = init)
 18. tidyselect:::walk_data_tree(new, data_mask, context_mask)
 19. tidyselect:::as_indices_sel_impl(...)
 20. tidyselect:::as_indices_impl(x, vars, strict = strict)
 21. tidyselect:::chr_as_locations(x, vars)
 22. vctrs::vec_as_location(x, n = length(vars), names = vars)
 24. vctrs:::stop_subscript_oob(...)
 25. vctrs:::stop_subscript(...)

**Comment by jmgirard** [comment]
Looks to be an issue where I updated `prep_data_cat()` but didn't update `cat_per_object()` to expect those changes. Will try to fix today.

**Comment by jmgirard** [comment]
Ok, it should be fixed in 1c540af62aeff0da163eb4b3abbb085ab1610de8

``` r
library(agreement)
data(ordered)
results1 <- cat_per_object(ordered, categories = c(0, 1, 2, 3), weighting = "linear")
results1
#> # A tibble: 20 × 3
#>    Object Weighting Agreement
#>     <int> <chr>         <dbl>
#>  1      1 linear        0.778
#>  2      2 linear        0.833
#>  3      3 linear        0.833
#>  4      4 linear        1    
#>  5      5 linear        1    
#>  6      6 linear        1    
#>  7      7 linear        0.667
#>  8      8 linear        0.556
#>  9      9 linear        1    
#> 10     10 linear        0.833
#> 11     11 linear        0.778
#> 12     12 linear        1    
#> 13     13 linear        0.833
#> 14     14 linear        0.8  
#> 15     15 linear        1    
#> 16     16 linear        1    
#> 17     17 linear        0.833
#> 18     18 linear        0.611
#> 19     19 linear        0.833
#> 20     20 linear        1
```

<sup>Created on 2022-09-08 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>
